### PR TITLE
support PHI columns for DataClass and SampleType

### PR DESCRIPTION
**Pull request by labkey-kevink**
#### Rationale
Teach DataClass and SampleType to respect PHI settings on columns similar to Lists.  Import/insert/update/delete URLs and permission depend on a user having PHI access to all columns on the table.

Introduce `PhiTransformedColumnInfo` marker interface for columns that have been transformed by PHI rules.  Query details API and DataRegion uses this to indicate which columns have been transformed due to PHI settings.  Query metadata indicates PHI transformed columns with a new `phiTransformed` property.  When PHI rules omit columns from a table, a DataRegion message is added to the grid to indicate one or more columns have been removed.

To allow an admin page within Biologics to change the folder's compliance settings, add the ability to change the settings in the ComplianceService along with a permissions check.

#### Related Pull Requests
* https://github.com/LabKey/compliance/pull/105
* https://github.com/LabKey/complianceActivities/pull/14
* https://github.com/LabKey/biologics/pull/833
* https://github.com/LabKey/gel/pull/6
* https://github.com/LabKey/nlp/pull/193

#### Changes
* DataClass and SampleType respect PHI rules
* Add `phiTransformed` property to query metadata for PHI transformed columns
* Add DataRegion message for PHI omitted columns
* Add ComplianceService methods to get and set compliance folder settings

### CODE REVIEW

File: api/src/org/labkey/api/data/QueryLogging.java [hunk 1]
```diff
@@ -164,6 +169,8 @@ public boolean isShouldAudit()
 
     public void setShouldAudit(boolean shouldAudit)
     {
+//        if (_readOnly)
+//            throw new IllegalStateException("This QueryLogging instance is read-only: " + _debugName);
```
Comment (labkey-adam) [inline api/src/org/labkey/api/data/QueryLogging.java:174]: Uncomment like above? Or delete?

File: experiment/src/org/labkey/experiment/api/LineageTreeDisplayColumn.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+//            ActionURL detailsURL = node.detailsURL();
+//            String name = node.getName();
+//
+//            StringBuilder sb = new StringBuilder();
+//            A(at(href, detailsURL), name).appendTo(sb);
+//            out.write(sb.toString());
+//
+//            renderLineageTree(ctx, out, lineage, node);
+//        }
+//        out.write("</ul>");
```
Comment (labkey-adam) [inline experiment/src/org/labkey/experiment/api/LineageTreeDisplayColumn.java]: Remove?

**Review by labkey-adam — CHANGES_REQUESTED** [review]

File: api/src/org/labkey/api/data/QueryLogging.java (hunk 1 above)
Comment (labkey-kevink) [reply api/src/org/labkey/api/data/QueryLogging.java:174]: I'd like to uncomment, but we are currently relying on the behavior.  Added a comment to the Issue for tracking.

**Comment by labkey-adam** [comment]
Looks like it needs a merge

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**